packages/ncids-js/src/components/nci-header/utils/megamenu.component.ts

```diff
@@ -0,0 +1,557 @@
+		let gridRowMenu = this.createDom('div', ['grid-row']);
+
+		// Go though each item in the response
+		[...data].map((item, index) => {
+			const mod = (index + 1) % 3;
+
+			// create the UL and container DIV
+			const MenuCol = this.createDom('div', ['usa-col', 'grid-gap-3']);
+			const MenuList = this.createDom('ul', ['nci-header-megamenu__list']);
+
+			// First link in list is Header link
+			const ParentMenuItem = this.createMenuLink(item.path, item.label);
+			MenuList.append(ParentMenuItem);
+			MenuCol.append(MenuList);
+			// Iterate over child items
+			[...item.children].map((item) => {
+				const MenuItem = this.createMenuLink(item.path, item.label);
+				MenuList.append(MenuItem);
+				MenuCol.append(MenuList);
+			});
+			gridRowMenu.append(MenuCol);
+			wrapper.append(gridRowMenu);
+			// if zero then we're at 3 and we should make a new row
+			if (mod === 0) {
+				gridRowMenu = this.createDom('div', ['grid-row']);
+			}
+		});
+
+		// If there is a leftover space then we need to fill
+		// it. Based on the amount of items in data, create
+		// an empty div and append to the last row container
+		if (leftover > 0) {
+			const MenuFiller = this.createDom('div', [
+				'usa-col',
+				leftover < 2 ? 'grid-gap-6' : 'grid-gap-3',
+			]);
+			gridRowMenu.append(MenuFiller);
+		}
+		return wrapper;
+	}
+
+	/**
+	 * Utility function to create the links used in the
+	 * MegaMenu Lists
+	 *
+	 * @param {string} path URL for menu link
+	 * @param {string} label Text menu label for link
+	 *
+	 * @return {Element} DIV layer HTML element
+	 * @private
+	 */
+	private createMenuLink(path: string, label: string): Element {
+		const MenuItem = this.createDom('li', ['nci-header-megamenu__item']);
+		const MenuLink = this.createDom('a', [], [{ href: path }]);
+		MenuLink.innerHTML = label;
+		MenuItem.append(MenuLink);
+		return MenuItem;
+	}
+
+	/**
+	 * Async call to get the data from the API
+	 *
+	 * @param {string} id of parent section.
+	 *
+	 * @return {Promise<Array<NavItem>>} HTML element
+	 * @private
+	 */
+	private async getMenuData(id: string): Promise<Array<NavItem>> {
+		const apiUrl = id;
+		const response = await fetch(apiUrl)
+			.then((response) => {
+				return response.json();
+			})
+			.catch((error) => {
+				console.error(`Fetch Error: ${id} ${error}`);
+				return [];
+			});
+		return await response;
+	}
+
+	/**
+	 * Returns array of LI elements in the primary navigation
+	 *
+	 * @return {NodeListOf<HTMLElement>}
+	 * @private
+	 */
+	private queryListSections(): HTMLElement[] {
+		const selector = this.options.listElementClass;
+		return Array.from(this.element.querySelectorAll(<string>selector));
+	}
+
+	/**
+	 * Returns the HTMLElement that contains the MegaMenu
+	 *
+	 * @return {<HTMLElement>} MegaMenu DOM Element
+	 * @private
+	 */
+	private queryMegaMenu(button: Element): HTMLElement {
+		const selector = '.nci-header-megamenu';
+		return <HTMLElement>button.parentElement?.querySelector(selector);
```

Comment:
let's make sure parentElement is not optional.

There are a couple ways to do this, but this works:

```typescript
    private queryMegaMenu(button: Element): HTMLElement {
		const selector = '.nci-header-megamenu';
		const parent = <HTMLElement>button.parentElement;
		return <HTMLElement>parent.querySelector(selector);
	}
```